packages/jest-matchers/src/index.js

```diff
@@ -10,27 +10,40 @@
 
 'use strict';
 
-import type {ThrowingMatcherFn, ExpectationResult} from '../types';
+import type {
+  Expect,
+  ExpectationResult,
+  ExpectationObject,
+  MatchersObject,
+  RawMatcherFn,
+  ThrowingMatcherFn,
+} from '../types';
 
 const matchers = require('./matchers');
+const GLOBAL_MATCHERS_OBJECT_SYMBOL = Symbol.for('$$jest-matchers-object');
 
-function expect(actual: any) {
+if (!global[GLOBAL_MATCHERS_OBJECT_SYMBOL]) {
+  Object.defineProperty(global, GLOBAL_MATCHERS_OBJECT_SYMBOL, {value: {}});
```

Comment:
@cpojer how about that?
we can always do a closure var too, it's not really a big difference :)